src/components/home/elements/AppWallet.vue

```diff
@@ -0,0 +1,229 @@
+                    v-model="coldPubKey"
+                    aria-describedby="inputLiveFeedback"
+                    placeholder="Please input public key of cold wallet">
+      </b-form-input>
+      <b-form-invalid-feedback id="inputLiveFeedback">
+        Invalid Public Key
+      </b-form-invalid-feedback>
+    </b-form-group>
+    <b-row class="row">
+      <b-col class="col-lef">
+      </b-col>
+      <b-col class="col-rit">
+        <b-button
+          block
+          class="btn-confirm"
+          variant="warning"
+          size="lg"
+          :disabled="isSubmitDisabled()"
+          @click="sendData">Confirm
+        </b-button>
+      </b-col>
+    </b-row>
+  </div>
+</template>
+
+<script>
+import crypto from '@/utils/crypto'
+import { PROTOCOL, API_VERSION, OPC_ACCOUNT } from '@/constants.js'
+import base58 from '@/libs/base58'
+export default {
+    name: 'LedgerWallet',
+    data: function() {
+        return {
+            coldAddress: '',
+            jsonObj: '',
+            qrInit: false,
+            paused: false,
+            coldPubKey: '',
+            opc: '',
+            api: '',
+            protocol: ''
+        }
+    },
+    props: {
+        address: {
+            type: String,
+            default: ''
+        }
+    },
+    methods: {
+        closeModal() {
+            this.$refs.appWalletModal.hide()
+        },
+        async onInit(promise) {
+            try {
+                this.qrInit = true
+                await promise
+            } catch (error) {
+                if (error.name === 'NotAllowedError') {
+                    throw Error('user denied camera access permission')
+                } else if (error.name === 'NotFoundError') {
+                    throw Error('no suitable camera device installed')
+                } else if (error.name === 'NotSupportedError') {
+                    throw Error('page is not served over HTTPS (or localhost)')
+                } else if (error.name === 'NotReadableError') {
+                    throw Error('maybe camera is already in use')
+                } else if (error.name === 'OverconstarinedError') {
+                    throw Error('pass constraints do not match any camera')
+                } else {
+                    throw Error('browser is probably lacking features(WebRTC, Canvas)')
+                }
+            } finally {
+                this.qrInit = false
+            }
+        },
+        onDecode: function(decodeString) {
+            this.paused = true
+            try {
+                this.jsonObj = JSON.parse(decodeString)
+                this.coldAddress = this.jsonObj.address
+                this.coldPubKey = this.jsonObj.publicKey
+                this.opc = this.jsonObj.opc
+                this.api = this.jsonObj.api
+                this.protocol = this.jsonObj.protocol
+            } catch (e) {
+                this.paused = false
+            }
+            if (!this.isValidAddress) {
```

Comment:
`this.isValidAddress` should change to `this.isValidColdAddress(this.coldAddress)`